test/files/run/t6028.check

```diff
@@ -1,7 +1,7 @@
 [[syntax trees at end of                lambdalift]] // newSource1.scala
 package <empty> {
   class T extends Object {
-    <paramaccessor> val classParam: Int = _;
+    <paramaccessor> var classParam: Int = _;
```

Comment:
Here's a minimization that shows why this changed:

```
abstract class C[@specialized(Int) T] {
  def f: T
}

class K {
  private[this] val x: Int = 1
  def foo: Int = {
    val c = new C[Int] { def f = x }
    c.f
  }
}
```

`x` is made `notPrivate`, i guess becuase it's accessed in the specialized method `f$mcI$sp`. Now `x` is also made non-final, even if it's not written outside the constructor.

Since this can affect code that doesn't directly use `@specialized` (code that only uses / extends a specialized class / trait), it would be good if we could avoid it. (Field `final`ity affects the memory model.)

Reply:
Ahh I see, good catch 👍